be/src/olap/olap_cond.cpp

```diff
@@ -681,17 +681,17 @@ int Conditions::delete_pruning_filter(const std::vector<KeyRange>& zone_maps) co
     for (auto& cond_it : _columns) {
         /*
          * this is base on the assumption that the delete condition
-         * is only about key field, not about value field.
+         * is only about key field, not about value field except the storage model is duplicate.
         */
-        if (cond_it.second->is_key() && cond_it.first > zone_maps.size()) {
+        if ((cond_it.second->is_key() || _schema->keys_type() == KeysType::DUP_KEYS) && cond_it.first > zone_maps.size()) {
```

Comment:
`cond_it.second->is_key() || _schema->keys_type() == KeysType::DUP_KEYS`

This condition shows many times. How about extract it to a method?